### PR TITLE
fix "build complete" message

### PR DESCRIPTION
**Pull request by tomerd**
motivation: print "build complete" message reliably

changes:
* move the logic to printing the "build complete" message to the build complete handler
* add tests

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/Build/BuildOperationBuildSystemDelegateHandler.swift [hunk 1]
```diff
@@ -638,13 +649,7 @@ fileprivate struct CommandTaskTracker {
...
             totalCount -= 1
         case .isComplete:
-            if (totalCount == finishedCount) {
-                let latestOutput: String? = latestFinishedText
-                latestFinishedText = """
-                \(latestOutput ?? "")\n
-                * Build Completed!
-                """
-            }
+            break
```
Comment (tomerd) [inline Sources/Build/BuildOperationBuildSystemDelegateHandler.swift:652]: was added in https://github.com/apple/swift-package-manager/pull/2971

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]
Looks good to me!

**Comment by abertelrud** [comment]
Linux smoke test failure looks spurious.

**Comment by abertelrud** [comment]
@swift-ci please smoke test linux

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Pull request merged**